homeassistant/helpers/area_registry.py

```diff
@@ -197,8 +197,19 @@ def async_get_area(self, area_id: str) -> AreaEntry | None:
 
     @callback
     def async_get_area_by_name(self, name: str) -> AreaEntry | None:
-        """Get area by name."""
-        return self.areas.get_by_name(name)
+        """Get area by name or alias."""
+        if area := self.areas.get_by_name(name):
+            return area
+
+        # Check aliases
+        normalized_name = normalize_name(name)
```

Comment:
This is not using an index and will be very inefficient. Shouldn't we add a secondary alias index so we can call `areas.get_by_alias(name)` ?